### PR TITLE
Fix version parsing

### PR DESCRIPTION
**Pull request by lnguyen1401**
fix the version parsing

### CODE REVIEW

File: pkg/utils/clusterversion.go [hunk 1]
```diff
@@ -73,7 +74,9 @@ func IsVersionHigherThan(input string) bool {
 		return false
 	}
 	// Handle the clusternames that have more than 4 chars(such as 4.10.0-rc.4)
-	shortVersion := version[0:4]
+	re := regexp.MustCompile("([0-9]+).([0-9]+)([0-9]?)")
+	shortVersion := re.FindString(version)
+	fmt.Println("Cluster shortVersion is: ", shortVersion)
```
Comment (boranx) [inline pkg/utils/clusterversion.go:79]: I think we can remove _fmt_

**Review by abyrne55 — APPROVED** [review]
/lgtm

**Comment by fahlmant** [comment]
Take a look at how this test is structured:
Here the test struct is created 
https://github.com/openshift/cloud-ingress-operator/blob/master/pkg/controller/publishingstrategy/publishingstrategy_controller_test.go#L680
And here each test in the list is ran
https://github.com/openshift/cloud-ingress-operator/blob/master/pkg/controller/publishingstrategy/publishingstrategy_controller_test.go#L778

**Comment by lnguyen1401** [comment]
restructured the test, but logically the same

**Review by fahlmant — APPROVED** [review]
This is awesome, thanks :)
/approve

**Review by abyrne55 — APPROVED** [review]
/lgtm again :)

**Pull request merged**